trunk/SUAVE/Components/Energy/Converters/Propeller.py

```diff
@@ -112,41 +114,356 @@ def spin(self,conditions):
         Cp                           [-] (coefficient of power)
 
         Properties Used:
-        self.prop_attributes.
-          number_blades                 [-]
-          tip_radius                    [m]
-          hub_radius                    [m]
-          twist_distribution            [radians]
-          chord_distribution            [m]
-          mid_chord_aligment            [m] (distance from the mid chord to the line axis out of the center of the blade)
-          lift_curve_slope              [1/radians] (2D lift curve slope of the airfoil)
-          cd_coefficients               [-]
-         drag_reference_reynolds_number [-]
-         reynolds_scaling_exponent      [-]
-        self.thrust_angle               [radians]
+        self. 
+          number_blades              [-]
+          tip_radius                 [m]
+          hub_radius                 [m]
+          twist_distribution         [radians]
+          chord_distribution         [m]
+          mid_chord_aligment         [m] (distance from the mid chord to the line axis out of the center of the blade)
+          thrust_angle               [radians]
         """         
            
         #Unpack    
-        B        = self.prop_attributes.number_blades
-        R        = self.prop_attributes.tip_radius
-        Rh       = self.prop_attributes.hub_radius
-        beta     = self.prop_attributes.twist_distribution
-        c        = self.prop_attributes.chord_distribution
-        cl_a     = self.prop_attributes.lift_curve_slope
-        cd_coeff = self.prop_attributes.cd_coefficients   
-        re_ref   = self.prop_attributes.drag_reference_reynolds_number
-        x_re     = self.prop_attributes.reynolds_scaling_exponent  
-           
+        B      = self.number_blades
+        R      = self.tip_radius
+        Rh     = self.hub_radius
+        beta   = self.twist_distribution
+        c      = self.chord_distribution
+        omega1 = self.inputs.omega 
+        a_sec  = self.airfoil_sections        
+        a_secl = self.airfoil_section_location        
+        rho    = conditions.freestream.density[:,0,None]
+        mu     = conditions.freestream.dynamic_viscosity[:,0,None]
+        Vv     = conditions.frames.inertial.velocity_vector
+        Vh     = self.induced_hover_velocity 
+        a      = conditions.freestream.speed_of_sound[:,0,None]
+        T      = conditions.freestream.temperature[:,0,None]
+        theta  = self.thrust_angle
+        tc     = .12 # Thickness to chord
```

Comment:
Should this be hardcoded?

Reply:
I changed so that it is now a property of the propeller i.e. "self.thickness_to_chord_ratio". Now we will use the thickness to chord ratio at 70% of the propeller radius. This is commonly used in literature